dispatcher/backend/src/routes/utils.py

```diff
@@ -97,14 +97,16 @@ def remove_s3_secrets(response: dict):
         else:
             if not isinstance(response[key], str) or "://" not in response[key]:
                 continue
-            url = urlparse(response[key])
-            response[key] = url._replace(
-                query="&".join(
-                    [
-                        f"{key}={value}"
-                        for key, values in parse_qs(url.query).items()
-                        if str(key).lower() not in ["keyid", "secretaccesskey"]
-                        for value in values
-                    ]
-                )
-            ).geturl()
+            for part in [part for part in response[key].split() if "://" in part]:
+                url = urlparse(part)
+                url = url._replace(
+                    query="&".join(
+                        [
+                            f"{key}={value}"
+                            for key, values in parse_qs(url.query).items()
+                            if str(key).lower() not in ["keyid", "secretaccesskey"]
```

Comment:
This does not only remove the secret but the params altogether right? This might mislead one into thinking the URL lacked the secret info. We should use our `SECRET_REPLACEMENT` as elsewhere

Reply:
Correct